File: BSAP_ProjectMonitor/BSGlobalClass.vb
Imports MySql.Data.MySqlClient
Namespace BurnSoft
    Public Class BSDatabase
        Public Conn As MySqlConnection
        Private Function MyConnectionString() As String
            Dim sAns As String = "Server=" & System.Configuration.ConfigurationManager.AppSettings("DB_HOST")
            sAns &= ";user id=" & System.Configuration.ConfigurationManager.AppSettings("DB_UID")
            sAns &= ";password=" & System.Configuration.ConfigurationManager.AppSettings("DB_PWD")
            sAns &= ";persist security info=true"
            sAns &= ";database=" & System.Configuration.ConfigurationManager.AppSettings("DB_NAME")
            Return sAns
        End Function
        Public Function ConnectDB() As Integer
            Dim iAns As Integer = 0
            Err.Clear()
            Try
                Dim sConnect As String = MyConnectionString()
                Conn = New MySqlConnection(sConnect)
                Conn.Open()
            Catch ex As Exception
                iAns = Err.Number
                Dim strForm As String = "BSGlobalClass.BSDatabase"
                Dim strSubFunc As String = "ConnectDB"
                Dim sMsg As String = " - ERROR - " & Err.Number & " - " & ex.Message.ToString
                Call LogError(sMsg, strForm & "." & strSubFunc)
            End Try
            Return iAns
        End Function
        Public Sub ConnExe(ByVal SQL As String)
            Try
                If ConnectDB() = 0 Then
                    Dim CMD As New MySqlCommand
                    CMD.CommandText = SQL
                    CMD.Connection = Conn
                    CMD.ExecuteNonQuery()
                    CMD.Connection.Close()
                    CMD = Nothing
                    Call CloseDB()
                End If
            Catch ex As Exception
                Dim strForm As String = "BSGlobalClass.BSDatabase"
                Dim strSubFunc As String = "ConnExe"
                Dim sMsg As String = " - ERROR - " & Err.Number & " - " & ex.Message.ToString
                Call LogError(sMsg, strForm & "." & strSubFunc)
            End Try
        End Sub
        Public Sub CloseDB()
            If Conn.State <> ConnectionState.Closed Then
                Conn.Close()
            End If
            Conn = Nothing
        End Sub
    End Class
    Public Class BSSqliteDatabase
        Public Conn As SQLite.SQLiteConnection
        Private Function MyConnectionString() As String
            Return "Data Source=bsap_client.db"
        End Function
        Public Function ConnectDB() As Integer
            Dim iAns As Integer = 0
            Err.Clear()
            Try
                Dim sConnect As String = MyConnectionString()
                Conn = New SQLite.SQLiteConnection(sConnect)
                Conn.Open()
            Catch ex As Exception
                iAns = Err.Number
                Dim strForm As String = "BSGlobalClass.BSSqliteDatabase"
                Dim strSubFunc As String = "ConnectDB"
                Dim sMsg As String = " - ERROR - " & Err.Number & " - " & ex.Message.ToString
                Call LogError(sMsg, strForm & "." & strSubFunc)
            End Try
            Return iAns
        End Function
        Public Sub ConnExe(SQL As String)
            Try
                If ConnectDB() = 0 Then
                    Dim CMD As New SQLite.SQLiteCommand
                    With CMD
                        .CommandText = SQL
                        .Connection = Conn
                        .ExecuteNonQuery()
                        .Connection.Close()
                    End With
                    CMD = Nothing
                    Call CloseDB()
                End If
            Catch ex As Exception
                Dim strForm As String = "BSGlobalClass.BSSQLiteDatabase"
                Dim strSubFunc As String = "ConnExe"
                Dim sMsg As String = " - ERROR - " & Err.Number & " - " & ex.Message.ToString
                Call LogError(sMsg, strForm & "." & strSubFunc)
            End Try
        End Sub
        Public Sub CloseDB()
            If Conn.State <> ConnectionState.Closed Then
                Conn.Close()
            End If
            Conn = Nothing
        End Sub
    End Class
End Namespace

File: BSAP_SubProcessMonitor/BSGlobalClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;
using System.Configuration;
using System.Data;

namespace BurnSoft
{
    class BSSqliteDatabase
    {
        private string MyConnectionString => String.Format(
            "Server={0};user id={1};password={2};persist security info=true;database={3}",
            ConfigurationManager.AppSettings["DB_HOST"], ConfigurationManager.AppSettings["DB_UID"],
            ConfigurationManager.AppSettings["DB_PWD"], ConfigurationManager.AppSettings["DB_NAME"]);
        /// <summary>
        /// Main Class Name for error dumping.
        /// </summary>
        private string ClassLocation => "BurnSoft.BSDatabase";

        /// <summary>
        /// General Error Message Format getting the ClassLocation property and appending it to the sLocation and also appended it 
        /// to the ex.message, this was done to allow you narrow down the location of were the error occured.
        /// </summary>
        /// <param name="sLocation">Sub or Function Name</param>
        /// <param name="ex">Exception</param>
        /// <returns>string Class.SubOrFunction - Error Message</returns>
        /// <example>serror = ErrorMessage("getDBVersion", ex);</example>
        /// <remarks>copy and fill in the blank - ErrorMessage("", ex);</remarks>
        private string ErrorMessage(string sLocation, Exception ex)
        {
            return $"{ClassLocation}.{sLocation} - {ex.Message}";
        }

        public MySqlConnection Conn;
        public int ConnectDB(out string errMsg)
        {
            int iAns = 0;
            errMsg = @"";
            try
            {
                Conn = new MySqlConnection(MyConnectionString);
                Conn.Open();
            }
            catch (Exception e)
            {
                errMsg = ErrorMessage("ConnectDB", e);
                iAns = System.Runtime.InteropServices.Marshal.GetExceptionCode();
            }
            return iAns;
        }

        public void CloseDB()
        {
            if (Conn.State != ConnectionState.Closed)
            {
                Conn.Close();
            }
            Conn = null;
        }

        public void ConnExe(string SQL, out string errMsg)
        {
            errMsg = @"";
            string errMsgConn = @"";
            try
            {
                if (ConnectDB(out errMsgConn) == 0)
                {
                    MySqlCommand CMD = new MySqlCommand();
                    CMD.CommandText = SQL;
                    CMD.Connection = Conn;
                    CMD.ExecuteNonQuery();
                    CMD.Connection.Close();
                    CMD = null;
                    CloseDB();
                }
                else
                {
                    throw new Exception(errMsgConn);
                }
            }
            catch (Exception e)
            {
                errMsg = ErrorMessage("ConnExe", e);
            }
        }
    }
}
